pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/RealtimeTableDataManager.java

```diff
@@ -268,56 +269,69 @@ public void addSegment(String segmentName, TableConfig tableConfig, IndexLoading
     LoaderUtils.reloadFailureRecovery(indexDir);
 
     boolean isLLCSegment = SegmentName.isLowLevelConsumerSegmentName(segmentName);
-    LLCSegmentName llcSegmentName = null;
-    PartitionUpsertMetadataManager partitionUpsertMetadataManager = null;
-    if (isLLCSegment) {
-      llcSegmentName = new LLCSegmentName(segmentName);
-      if (_tableUpsertMetadataManager != null) {
-        partitionUpsertMetadataManager =
-            _tableUpsertMetadataManager.getOrCreatePartitionManager(llcSegmentName.getPartitionId());
+    if (indexDir.exists()) {
+      // Segment already exists on disk
+      if (realtimeSegmentZKMetadata.getStatus() == Status.DONE) {
+        // Metadata has been committed, load the local segment
+        try {
+          addSegment(ImmutableSegmentLoader.load(indexDir, indexLoadingConfig, schema));
+          return;
+        } catch (Exception e) {
+          if (isLLCSegment) {
+            // For LLC and segments, delete the local copy and download a new copy from the controller
+            FileUtils.deleteQuietly(indexDir);
+            if (e instanceof V3RemoveIndexException) {
+              _logger.info("Unable to remove index from V3 format segment: {}, downloading a new copy", segmentName, e);
```

Comment:
Why do we need to log the exception here?

Reply:
The exception contains the column that cannot be removed for debugging purpose. Here I just keep the same behavior as the offline one in `SegmentFetcherAndLoader`.